### PR TITLE
Add support for nix-direnv to direnv

### PR DESCRIPTION
**Pull request by DamienCassou**
### Description

Add support for [nix-direnv](https://github.com/nix-community/nix-direnv) from @Mic92 to the direnv module.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/rycee/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

### CODE REVIEW

**Comment by DamienCassou** [comment]
The build failure doesn't seem related to my PR:

```
trying https://gitlab.com/api/v4/projects/rycee%2Fnmt/repository/archive.tar.gz?sha=8e130d655ec396ce165763c95bbf4ac429810ca8

unpacking source archive /tmp/nix-build-source.drv-0/archive.tar.gz?sha=8e130d655ec396ce165763c95bbf4ac429810ca8

error: unknown hash algorithm '', at /nix/store/fhksmnm8dk9b1hs99sfzha4nwvpfs86n-nixpkgs-20.09pre227892.b27a19d5bf7/nixpkgs/pkgs/build-support/fetchurl/default.nix:119:3

(use '--show-trace' to show detailed location information)
```

File: modules/programs/direnv.nix [hunk 1]
```diff
@@ -78,8 +83,14 @@ in {
...
 
-    xdg.configFile."direnv/direnvrc" =
-      mkIf (cfg.stdlib != "") { text = cfg.stdlib; };
+    xdg.configFile."direnv/direnvrc" = let
+      stdlib = cfg.stdlib;
+      nixDirenv = cfg.enableNixDirenvIntegration;
+    in mkIf (cfg.stdlib != "" || nixDirenv) {
+      text = (optionalString (stdlib != "") (stdlib + "\n"))
+        + (optionalString nixDirenv
+          "source ${pkgs.nix-direnv}/share/nix-direnv/direnvrc");
```
Comment (Mic92) [inline modules/programs/direnv.nix:92]: Looks good to me.

**Comment by rycee** [comment]
Thanks! I rebased this with a few minor edits in b33802ca7f0675fa8f64dd0f322ab7a57e45ba6a.

**Comment by rileyinman** [comment]
Any reason this isn't enabled in the `release-20.03` branch? It looks like nix-direnv is available in the 20.03 nixpkgs branch.

**Comment by cole-h** [comment]
Probably because nobody expressed interest in it for 20.03 (until you). At the bottom of the README, it says, `If you need a module to be backported, then feel free to open an issue.` While more a feature of a module than an entire module, you can file an issue for it, or you can backport it yourself by making a PR against said release branch. I don't imagine this change being controversial.

**Comment by rileyinman** [comment]
Fair enough, I blame myself for not paying close enough attention :sweat_smile:

**Comment by cole-h** [comment]
No worries, it's easy to miss (quite literally the last sentence of the README). I don't expect many people have read the entire README, word-for-word, from start to finish.

**Pull request closed without merging**